pandas/tests/indexes/timedeltas/test_insert.py

```diff
@@ -82,6 +82,17 @@ def test_insert_invalid_na(self):
         with pytest.raises(TypeError, match="incompatible label"):
             idx.insert(0, np.datetime64("NaT"))
 
+    @pytest.mark.parametrize(
+        "value", [0, np.int64(0), np.float64(0), np.array(0), np.datetime64(456, "us")]
+    )
+    def test_insert_mismatched_types(self, value):
```

Comment:
unwritten convention

```suggestion
    def test_insert_mismatched_types_raises(self, value):
```